### ISSUE TITLE
$ENV{FORCE_FILTER_SIGNATURES} gives warning with empty ()

### ISSUE THREAD

**Issue by devel-chm** [Corion/filter-signatures#1]
If I force the use of Filter::signatures in Helpers.pm with
```perl
BEGIN {
   # force use of Filter::signatures
   no warnings 'experimental::signatures';
   $ENV{FORCE_FILTER_SIGNATURES} = 1;
}
```
And then
```perl
use Filter::signatures;
no warnings 'experimental::signatures';
use feature 'signatures';
```
I get the following warnings printed

>  perl -c Helpers.pm
The signatures feature is experimental at Helpers.pm line 122.
The signatures feature is experimental at Helpers.pm line 128.
Helpers.pm syntax OK

where the routines on lines 122 and 128 have parens with nothing inside:
```perl
sub glGetVersion_p() {
    my $glVersion = glGetString(GL_VERSION);
    ($glVersion) = ($glVersion =~ m!^(\d+\.\d+)!g);
    $glVersion
}

sub croak_on_gl_error() {
    my $error = glGetError();
    if( $error != GL_NO_ERROR ) {
        croak $glErrorStrings{ $error } || "Unknown OpenGL error: $error"
    };
}
```
If I don't set the environment variable, then no warnings are printed.

**Comment by Corion** [comment]
Whoops - that's what I get for only testing with the native Perl implementation
instead of also properly (resp. only) testing with the filter enabled. The above
two changes fix the issue for me. If they work for you as well, I'll upload them to CPAN soonish.

**Comment by Corion** [comment]
I've just released v0.07 with the (hopefully) fix for this onto CPAN.

**Comment by devel-chm** [comment]
I didn't get the warnings but I did get lots and lots of:

> Subroutine  called with parameters. at scripts/shadertoy.pl line 643.
Subroutine  called with parameters. at scripts/shadertoy.pl line 758.
Subroutine  called with parameters. at scripts/shadertoy.pl line 643.
Subroutine  called with parameters. at scripts/shadertoy.pl line 758.
Subroutine  called with parameters. at scripts/shadertoy.pl line 643.
Subroutine  called with parameters. at scripts/shadertoy.pl line 758.
Subroutine  called with parameters. at scripts/shadertoy.pl line 735.
Subroutine DESTROY called with parameters. at /cygdrive/e/chm/pogl/app-shadertoy/blib/lib/OpenGL/Shader/OpenGL4.pm line 82.
Subroutine DESTROY called with parameters. at /cygdrive/e/chm/pogl/app-shadertoy/blib/lib/OpenGL/Shader/OpenGL4.pm line 82 during global destruction.

until I quit the application.  Is this some leftover debug stuff or do I need to re-pull app-shadertoy?

**Comment by Corion** [comment]
No, this is likely a mistake that I didn't clean up.

Am 18. Dezember 2016 15:39:25 MEZ, schrieb devel-chm <notifications@github.com>:
>I didn't get the warnings but I did get lots and lots of:
>
>> Subroutine  called with parameters. at scripts/shadertoy.pl line 643.
>Subroutine  called with parameters. at scripts/shadertoy.pl line 758.
>Subroutine  called with parameters. at scripts/shadertoy.pl line 643.
>Subroutine  called with parameters. at scripts/shadertoy.pl line 758.
>Subroutine  called with parameters. at scripts/shadertoy.pl line 643.
>Subroutine  called with parameters. at scripts/shadertoy.pl line 758.
>Subroutine  called with parameters. at scripts/shadertoy.pl line 735.
>Subroutine DESTROY called with parameters. at
>/cygdrive/e/chm/pogl/app-shadertoy/blib/lib/OpenGL/Shader/OpenGL4.pm
>line 82.
>Subroutine DESTROY called with parameters. at
>/cygdrive/e/chm/pogl/app-shadertoy/blib/lib/OpenGL/Shader/OpenGL4.pm
>line 82 during global destruction.
>
>until I quit the application.  Is this some leftover debug stuff or do
>I need to re-pull app-shadertoy?
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/Corion/filter-signatures/issues/1#issuecomment-267824817

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

**Comment by Corion** [comment]
This should be fixed by v0.08 of Filter::signatures, which reinstates signature-less functions

https://github.com/Corion/filter-signatures/commit/2ff4a2dc196415254306b1ab6331badbf19f875d
